Iterable/iterable-web-sdk
# [MOB-8425]: strengthen network interceptors

## JIRA Ticket(s) if any

* [MOB-8425](https://iterable.atlassian.net/browse/MOB-8425)

## Description

Our network interceptors are prone to misses whenever new routes are added.

This PR makes it so that all places in the SDK calling our API are pulling from a consistent route object and that same route object is being used by the interceptors. This way, any new routes added to there will have to be made sure to work with our interceptors. 

## Test Steps
Start sample app
Test all API calls and make sure userID and email are properly attached. 

[MOB-8425]: https://iterable.atlassian.net/browse/MOB-8425?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
